src/Compilers/Core/Portable/Generated/Operations.xml.Generated.cs

```diff
@@ -4597,6 +4605,142 @@ internal sealed partial class LazyDynamicObjectCreationExpression : BaseDynamicO
+    {
+        private readonly Lazy<IOperation> _lazyExpression;
+        private readonly Lazy<ImmutableArray<IOperation>> _lazyArguments;
+        public LazyDynamicInvocationExpression(Lazy<IOperation> expression, ImmutableArray<ISymbol> applicableSymbols, Lazy<ImmutableArray<IOperation>> arguments, ImmutableArray<string> argumentNames, ImmutableArray<RefKind> argumentRefKinds, SemanticModel semanticModel, SyntaxNode syntax, ITypeSymbol type, Optional<object> constantValue, bool isImplicit) :
+            base(applicableSymbols, argumentNames, argumentRefKinds, semanticModel, syntax, type, constantValue, isImplicit)
+        {
+            _lazyExpression = expression ?? throw new System.ArgumentNullException(nameof(expression));
+            _lazyArguments = arguments ?? throw new System.ArgumentNullException(nameof(arguments));
+        }
+        protected override IOperation ExpressionImpl => _lazyExpression.Value;
+        protected override ImmutableArray<IOperation> ArgumentsImpl => _lazyArguments.Value;
+    }
+
+    /// <remarks>
+    /// Represents a dynamic indexer expression in C#.
+    /// </remarks>
+    internal abstract partial class BaseDynamicIndexerAccessExpression : HasDynamicArgumentsExpression, IDynamicIndexerAccessExpression
+    {
+        public BaseDynamicIndexerAccessExpression(ImmutableArray<ISymbol> applicableSymbols, ImmutableArray<string> argumentNames, ImmutableArray<RefKind> argumentRefKinds, SemanticModel semanticModel, SyntaxNode syntax, ITypeSymbol type, Optional<object> constantValue, bool isImplicit) :
+            base(OperationKind.DynamicIndexerAccessExpression, applicableSymbols, argumentNames, argumentRefKinds, semanticModel, syntax, type, constantValue, isImplicit)
+        {
+        }
+        protected abstract IOperation ExpressionImpl { get; }
+
+        public override IEnumerable<IOperation> Children
+        {
+            get
+            {
+                yield return Expression;
+                foreach (var argument in Arguments)
+                {
+                    yield return argument;
+                }
+            }
+        }
+        /// <summary>
+        /// Dynamically indexed expression.
+        /// </summary>
+        public IOperation Expression => Operation.SetParentOperation(ExpressionImpl, this);
+        public override void Accept(OperationVisitor visitor)
+        {
+            visitor.VisitDynamicIndexerAccessExpression(this);
+        }
+        public override TResult Accept<TArgument, TResult>(OperationVisitor<TArgument, TResult> visitor, TArgument argument)
+        {
+            return visitor.VisitDynamicIndexerAccessExpression(this, argument);
+        }
+    }
+
+    /// <remarks>
+    /// Represents a dynamic indexer expression in C#.
+    /// </remarks>
+    internal sealed partial class DynamicIndexerAccessExpression : BaseDynamicIndexerAccessExpression, IHasDynamicArgumentsExpression, IDynamicIndexerAccessExpression
```

Comment:
Are there any consumers of this class other than `OperationCloner`? (Can this class be removed?)

Should `OperationCloner` defer to the `IOperation` implementation to do the cloning? That is, should there be an `Operation.Clone()` method where `LazyDynamicIndexerAccessExpression.Clone()` returns a `LazyDynamicIndexerAccessExpression` rather than a `DynamicIndexerAccessExpression`?

Reply:
Never mind, since the clone needs to be a deep clone, returning a `LazyDynamicIndexerAccessExpression` would mean lazily cloning each of the fields, which seems unnecessarily complicated.